test/testlib.py

```diff
@@ -429,20 +430,32 @@ class MachineCase(unittest.TestCase):
     machine = None
     machine_class = None
     browser = None
-    machines = [ ]
+
+    def __init__(self, *args, **kwargs):
+        super(MachineCase, self).__init__(*args, **kwargs)
+        self.machines = []
 
     def label(self):
         (unused, sep, label) = self.id().partition(".")
         return label.replace(".", "-")
 
     def new_machine(self, image=None):
-        if not self.machine_class:
-            import testvm
-            self.machine_class = testvm.VirtMachine
-        m = self.machine_class(verbose=arg_trace, image=image, label=self.label())
-        self.addCleanup(lambda: m.kill())
-        self.machines.append(m)
-        return m
+        import testvm
+        machine_class = self.machine_class
+        if arg_address:
+            if machine_class:
+                raise unittest.SkipTest("Cannot run this test when specific machine address is specified")
+            if len(self.machines) != 0:
+                raise unittest.SkipTest("Cannot run multiple machines a specific machine address is specified")
```

Comment:
I suggest:
"Cannot run multiple machines `if` a specific machine address is specified"